lib/NetPacksLib.cpp

```diff
@@ -240,6 +240,15 @@ DLL_LINKAGE void SetAvailableHeroes::applyGs( CGameState *gs )
 	for (int i = 0; i < GameConstants::AVAILABLE_HEROES_PER_PLAYER; i++)
 	{
 		CGHeroInstance *h = (hid[i]>=0 ?  gs->hpool.heroesPool[hid[i]].get() : nullptr);
+		for (auto I = h->artifactsInBackpack.begin(), E = h->artifactsInBackpack.end();
+		     I != E; ++I)
+		{
+			if (I->artifact->artType->id == ArtifactID::GRAIL)
+			{
+				h->artifactsInBackpack.erase(I);
+				break;

```

Comment:
"subsequent iterations" are not absolutely useless, in theory there might be more than one grail. "Expanded" inline function is less readable, but 

```cpp
vsdt::erase_if(artifactsInBackpack, [](const ArtSlotInfo& a) 
{ 
return a.artifact->artType->id == ArtifactID::GRAIL; 
});
```
Is shorter and well readable.